src/test/rpc/NoRipple_test.cpp

```diff
@@ -72,45 +72,90 @@ class NoRipple_test : public beast::unit_test::suite
         testcase("Set noripple on a line with negative balance");
 
         using namespace jtx;
-        Env env(*this, features);
-
         auto const gw = Account("gateway");
         auto const alice = Account("alice");
         auto const bob = Account("bob");
         auto const carol = Account("carol");
 
-        env.fund(XRP(10000), gw, alice, bob, carol);
+        // fix1578 changes the return code.  Verify expected behavior
+        // without and with fix1578.
+        for (auto const tweakedFeatures :
+            {features - fix1578, features | fix1578})
+        {
+            Env env(*this, tweakedFeatures);
 
-        env.trust(alice["USD"](100), bob);
-        env.trust(bob["USD"](100), carol);
-        env.close();
+            env.fund(XRP(10000), gw, alice, bob, carol);
 
-        env(pay(alice, carol, carol["USD"](50)), path(bob));
+            env.trust(alice["USD"](100), bob);
+            env.trust(bob["USD"](100), carol);
+            env.close();
 
-        env(trust(alice, bob["USD"](100), bob, tfSetNoRipple));
-        env(trust(bob, carol["USD"](100), carol, tfSetNoRipple));
-        env.close();
+            // After this payment alice has a -50 USD balance with bob, and
+            // bob has a -50 USD balance with carol.  So neither alice nor
+            // bob should be able to clear the noRipple flag.
+            env(pay(alice, carol, carol["USD"](50)), path(bob));
+            env.close();
 
-        Json::Value params;
-        params[jss::source_account] = alice.human();
-        params[jss::destination_account] = carol.human();
-        params[jss::destination_amount] = [] {
-            Json::Value dest_amt;
-            dest_amt[jss::currency] = "USD";
-            dest_amt[jss::value] = "1";
-            dest_amt[jss::issuer] = Account("carol").human();
-            return dest_amt;
-        }();
+            TER const terNeg {tweakedFeatures[fix1578] ?
+                TER {tecNO_PERMISSION} : TER {tesSUCCESS}};
 
-        auto const resp = env.rpc("json", "ripple_path_find", to_string(params));
-        BEAST_EXPECT(resp[jss::result][jss::alternatives].size()==1);
+            env(trust(alice, bob["USD"](100), bob,   tfSetNoRipple), ter(terNeg));
+            env(trust(bob, carol["USD"](100), carol, tfSetNoRipple), ter(terNeg));
```

Comment:
Nit: Long lines

Reply:
Personal preference: I'm not overly worried about going over by a couple of characters.